secrethub/resource_service.go

```diff
@@ -0,0 +1,106 @@
+package secrethub
+
+import (
+	"github.com/secrethub/secrethub-go/internals/api"
+	"github.com/secrethub/secrethub-go/pkg/secrethub/credentials"
+
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceService() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceServiceCreate,
+		Read:   resourceServiceRead,
+		Delete: resourceServiceDelete,
+		Schema: map[string]*schema.Schema{
+			"repo": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The path of the repository on which the service operates.",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "A description of the service so others will recognize it.",
+			},
+			"credential": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Sensitive:   true,
+				Description: "The credential of the service account.",
+			},
+			"id": {
```

Comment:
> id is a reserved field name

Shall we use `service_id` instead? We also use that in the API.

Reply:
Ahh nevermind, I think we can just drop the field, as you can access the id set with `SetId`